python/dgl/transforms/functional.py

```diff
@@ -3300,4 +3302,69 @@ def norm_by_dst(g, etype=None):
 
     return norm
 
+def rwpe(g, k):
+    r"""Return random walk positional encodings.
+
+    For internal use.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The homogeneous graph.
+    k : int
+        The number of random walk steps.
+
+    Returns
+    -------
+    Tensor
+        The random walk positional encodings.
+    """
+    # get 1-step random walk probabilities as A * D^-1
+    A = g.adj(scipy_fmt='csr') # adjacency matrix
+    Dinv = sparse.diags(F.asnumpy(g.in_degrees()).clip(1) ** -1.0, dtype=float) # D^-1
+    RW = A * Dinv
+
+    # Iterate for k steps
+    PE = [F.tensor(RW.diagonal()).float()]
+    RW_power = RW
+    for _ in range(k-1):
+        RW_power = RW_power * RW
+        PE.append(F.tensor(RW_power.diagonal()).float())
+    PE = F.stack(PE,dim=-1)
+    
+    return PE
+
+def laplacian_pe(g, k):
+    r"""Return laplacian positional encodings.
+
+    For internal use.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The homogeneous graph.
+    k : int
+        Number of smallest non-trivial eigenvectors to use for positional encoding.
+
+    Returns
+    -------
+    Tensor
+        The laplacian positional encodings.
+    """
+    # get laplacian matrix as I - D^-0.5 * A * D^-0.5
+    A = g.adj(scipy_fmt='csr') # adjacency matrix
+    N = sparse.diags(F.asnumpy(g.in_degrees()).clip(1) ** -0.5, dtype=float) # D^-1/2
+    L = sparse.eye(g.number_of_nodes()) - N * A * N
+
+    # select eigenvectors with smaller eigenvalues
+    EigVal, EigVec = np.linalg.eig(L.toarray())
+    idx = EigVal.argsort() # increasing order
+    EigVal, EigVec = EigVal[idx], np.real(EigVec[:,idx])
+
+    # get random flip signs
```

Comment:
Can we add an argument to make this step optional?

Reply:
It seems that the random flip operation is necessary during training. If we do not do it here, the customers have to manually add it during the training iteration. How do you think about it?

Reply:
Ok. Then let's keep it.